tests/test_add_start_node.py

```diff
@@ -0,0 +1,38 @@
+from selenium.webdriver.remote.webdriver import WebDriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver import ActionChains
+from selenium.common.exceptions import NoSuchElementException
+
+web = "https://machinist.flapflap.io/"
+
+
+def test_add_node(driver: WebDriver):
+
+    driver.get(web)
+    source = driver.find_element(By.XPATH, '//*[@id="__next"]/main/div[1]')
+    action = ActionChains(driver)
+    action.double_click(source).perform()
+
+    # check if node is added by verifying if there is text q1
+    inspect_val = driver.find_element(
+        By.XPATH, '//*[@id="__next"]/main/div[1]/div/div[2]/p'
+    ).text
+    true_val = "q1"
+
+    assert inspect_val == true_val
+
+
+def test_start_node(driver: WebDriver):
+    driver.get(web)
+    driver.find_element(By.XPATH, '//*[@id="__next"]/main/div[1]/div/div').click()
+    driver.find_element(
+        By.XPATH, '//*[@id="__next"]/main/section/div/div[3]/div[1]/label[1]/span[1]'
+    ).click()
```

Comment:
I think right now you are randomly clicking on the canvas but to improve the test you can specify a location(x,y) where you want to create a node. I think this might help a bit
https://stackoverflow.com/questions/6775351/clicking-at-coordinates-without-identifying-element/42152313 I ran into the same prob and I am trying to figure it out as well